source/code/omi_preexec/omi-mysql.cpp

```diff
@@ -82,113 +83,19 @@ static int MySQL_ConfigurationParser(uid_t uid, gid_t gid, const std::wstring& c
         return 0;
     }
 
-    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Parsing MySQL configuration file: ", cnfFile));
+    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Loading MySQL configuration file: ", cnfFile));
 
-    std::vector<std::wstring> lines;
-    try
+    MySQL::MySQL_ConfigurationScanner configScanner;
+    if (!configScanner.Scan(cnfFile, auth))
     {
-        SCXStream::NLFs nlfs;
-        SCXFile::ReadAllLines(cnfFile, lines, nlfs );
-    }
-    catch (const SCXException& e)
-    {
-        SCX_LOGERROR(hLog, StrAppend(L"Error reading configuration file: ", e.What()));
+        SCX_LOGERROR(hLog, StrAppend(L"Error reading ", cnfFile));
```

Comment:
The class is not throwing exceptions - all the exceptions are already logged and indicated. This log statement is an additional log on top of an exception.
 I will get into the reason for not throwing exception from the class in the next comment.